File: chapter2-Μπαρμπούτι/ex2.2.rb
puts("Δώσε τον αριθμό εκκίνησης ")
a=gets.chomp.to_i

step=1
m=a
while a!=1
    if a%2==1
        a=a*3+1
    else
        a=a/2
    end

    puts(a)
    if a>m
        m=a
    end
    step = step+1
end

puts("Πλήθος βημάτων", step)
puts("Μέγιστος αριθμός το ", m)

File: chapter2/ex2.2.py
a=int(input("Δώσε τον αριθμό εκκίνησης "))

step=1
m=a
while a!=1:
    if a%2==1:
        a=a*3+1
    else:
        a=a/2

    print(a)
    if a>m:
        m=a
    step = step+1

print("Πλήθος βημάτων", step)
print("Μέγιστος αριθμός το ", m)
